app/models/form.rb

```diff
@@ -265,16 +275,31 @@ def current_version
     versions.current.first
   end
 
+  def oldest_accepted_version_id
+    return @oldest_accepted_version_id if defined?(@oldest_accepted_version_id)
+    versions.oldest_accepted.first&.id
+  end
+
+  def oldest_accepted_version
+    versions.find(oldest_accepted_version_id) if oldest_accepted_version_id.present?
+  end
+
+  def update_oldest_accepted
+    if defined?(@oldest_accepted_version_id) &&
+        @oldest_accepted_version_id != versions.oldest_accepted.first.id
+      versions.oldest_accepted.first.update!(is_oldest_accepted: false)
+      versions.find(@oldest_accepted_version_id).update!(is_oldest_accepted: true)
```

Comment:
If you do as I suggest above, this could be:

```
  if defined?(@oldest_accepted_version_id)
    persisted_oldest_version_accepted&.update!(is_oldest_accepted: false)
    oldest_version_accepted.update!(is_oldest_accepted: true)
  end
```

Right? Rails won't do the update on the persisted one if it's already false.